src/form/form.js

```diff
@@ -0,0 +1,669 @@
+         * @returns {Object}
+         */
+        getRanges() {
+            return this.getConfig().ranges || {};
+        },
+
+        /**
+         * Gets the title set to the form.
+         * @returns {String}
+         */
+        getTitle() {
+            return this.getConfig().title;
+        },
+
+        /**
+         * Change the title of the form
+         * @param {String} title
+         * @returns {form}
+         * @fires titlechange after the title has been changed
+         */
+        setTitle(title) {
+            this.getConfig().title = title;
+
+            if (this.is('rendered')) {
+                controls.$title.text(title);
+                hider.toggle(controls.$title, !!title);
+            }
+
+            /**
+             * @event titlechange
+             * @param {String} title
+             */
+            this.trigger('titlechange', title);
+
+            return this;
+        },
+
+        /**
+         * Gets a widget by its uri
+         * @param {String} uri
+         * @returns {widgetForm}
+         */
+        getWidget(uri) {
+            return widgets.get(uri) || null;
+        },
+
+        /**
+         * Adds a widget to the form
+         * @param {widgetConfig} definition
+         * @returns {Promise<widgetForm>}
+         * @throws TypeError if the widget definition is invalid
+         * @fires change when the widget's value changes
+         * @fires change-<uri> when the widget's value changes
+         * @fires widgetadd after the widget has been added
+         */
+        addWidget(definition) {
+            return validateDefinition(this, definition, 'uri')
+                .then(() => {
+                    const ranges = this.getRanges();
+                    if (definition.range && 'string' === typeof definition.range) {
+                        definition.range = ranges[definition.range];
+                    }
+
+                    if (!definition.widget) {
+                        definition.widget = widgetDefinitions.DEFAULT;
+                    }
+
+                    return new Promise(resolve => {
+                        const widget = widgetFactory(controls.$widgets, definition);
+                        widgets.set(definition.uri, widget);
+                        widget
+                            .on('change.form', value => {
+                                /**
+                                 * @event change
+                                 * @param {String} uri
+                                 * @param {String} value
+                                 */
+                                this.trigger('change', definition.uri, value);
+
+                                /**
+                                 * @event change-<uri>
+                                 * @param {String} value
+                                 */
+                                this.trigger(`change-${definition.uri}`, value);
```

Comment:
what's the purpose of this event ?

Reply:
The purpose is to give the ability to listen change on a particular widget.